explorer/interpreter/interpreter.cpp

```diff
@@ -966,21 +1020,28 @@ auto Interpreter::CallDestructor(Nonnull<const DestructorDeclaration*> fun,
     return ProgramError(fun->source_loc())
            << "destructors currently don't support `addr self` bindings";
   }
-  if (placeholder->value_node().has_value()) {
-    method_scope.Bind(*placeholder->value_node(), receiver);
+  if (auto& value_node = placeholder->value_node()) {
+    if (value_node->expression_category() == ExpressionCategory::Value) {
+      method_scope.BindValue(*placeholder->value_node(), receiver);
+    } else {
+      // TODO: [self addr: Self*] destructors not implemented yet.
+      CARBON_FATAL() << "[self addr: Self*] destructors not implemented yet";
```

Comment:
```suggestion
      CARBON_FATAL() << "TODO: [self addr: Self*] destructors not implemented yet";
```

It feels odd to me to have a TODO that copies the error message, can just add the TODO to the message. :)